ddl/ddl_tiflash_api.go

```diff
@@ -463,6 +511,125 @@ func (d *ddl) pollTiFlashReplicaStatus(ctx sessionctx.Context, pollTiFlashContex
 	return allReplicaReady, nil
 }
 
+func (d *ddl) pollTiFlashPeerInfo(ctx sessionctx.Context, pollTiFlashContext *TiFlashManagementContext) (bool, error) {
+	allReplicaReady := true
+
+	if len(pollTiFlashContext.TiFlashStores) == 0 {
+		return true, nil
+	}
+
+	// Start to process every table.
+	schema := d.GetInfoSchemaWithInterceptor(ctx)
+	if schema == nil {
+		return false, errors.New("Schema is nil")
+	}
+
+	var tableMap = make(map[int64]TiFlashReplicaStatus)
+	// Collect TiFlash Replica info, for every table.
+	for _, db := range schema.AllSchemas() {
+		tbls := schema.SchemaTables(db.Name)
+		for _, tbl := range tbls {
+			tblInfo := tbl.Meta()
+			GetTiFlashReplicaMapInfo(tblInfo, &tableMap)
+		}
+	}
+
+	tiflashStoreIDs := make([]int64, len(pollTiFlashContext.TiFlashStores))
+
+	for i, store := range pollTiFlashContext.TiFlashStores {
+		tiflashStoreIDs[i-1] = store.Store.ID
+	}
+
+	sort.Slice(tiflashStoreIDs, func(i, j int) bool { return tiflashStoreIDs[i] < tiflashStoreIDs[j] })
+
+	isTiflashPeer := func(tiflashPeerIDs []int64, peerID int64) bool {
+		for _, tiflashPeerID := range tiflashPeerIDs {
+			if tiflashPeerID == peerID {
+				return true
+			}
+		}
+
+		return false
+	}
+
+	isTiflashAllPeerDown := func(tiflashPeerIDs []int64, tiflashDownPeerIDs []int64) bool {
+		return reflect.DeepEqual(tiflashPeerIDs, tiflashDownPeerIDs)
+	}
+
+	removeDuplicate := func(peerIDs []int64) []int64 {
+		result := make([]int64, 0, len(peerIDs))
+		temp := map[int64]struct{}{}
+		for _, item := range peerIDs {
+			if _, ok := temp[item]; !ok {
+				temp[item] = struct{}{}
+				result = append(result, item)
+			}
+		}
+		return result
+	}
+
+	var invaildRegions helper.RegionsInfo
+
+	if err := infosync.GetTiFlashPDInvalidRegionsInfo(context.Background(), &invaildRegions); err != nil {
```

Comment:
I'm afraid that will be a lot of regions.

Reply:
I wonder whether we can do this check on PD, instead of TiDB.

Reply:
Because the "down|pending" peer info, TIDB won't store it, need request to PD ask for it.

Reply:
If we do this job in PD, PD can't see any `schema` or `table` info in that. we also can't get any info about current `TiFlash peer info`.